### PR TITLE
Fix edge port usages

### PR DESCRIPTION
**Pull request by joe4dev**
Depends on https://github.com/localstack/localstack/pull/9600

## Motivation

We previously removed the write-usage of `EDGE_PORT` but kept read-access (e.g., `config.EDGE_PORT`) in our codebase. Given the many occurrences, it was easy to miss a few write-usages, and read-access gets mixed up with internal vs. external access.
I just noticed some of these issues while reviewing https://github.com/localstack/localstack/pull/9600 

## Changes

This PR fixes these inconsistencies by migrating all usages.

* Remove a few `EDGE_PORT` write usages we previously missed
* Fix some mixed-up `EDGE_PORT` usages migrating them to the new networking API
* Migrate usages of `LOCALSTACK_HOSTNAME` and `EDGE_PORT` in Lambda functions to `AWS_ENDPOINT_URL`. They are functionally equivalent.

## Testing

I have already run the full pipeline with these changes on top of the commits from https://github.com/localstack/localstack/pull/9600 All tests were green except for one flake in community `test_firehose_kinesis_to_s3`. Subsequently, I rebased upon master for a clean PR view.

### CODE REVIEW

File: bin/docker-entrypoint.sh [hunk 1]
```diff
@@ -18,11 +18,9 @@ if [[ $LOCALSTACK_API_KEY ]] && ! compgen -G /usr/lib/localstack/.*pro-version >
     echo ""
 fi
 
-# Strip `LOCALSTACK_` prefix in environment variables name (except
-# LOCALSTACK_HOST and LOCALSTACK_HOSTNAME)
+# Strip `LOCALSTACK_` prefix in environment variables name (except LOCALSTACK_HOST)
 source <(
   env |
-  grep -v -e '^LOCALSTACK_HOSTNAME' |
```
Comment (joe4dev) [inline bin/docker-entrypoint.sh:25]: or should we keep it here to get deprecation warnings in the container?

File: localstack/utils/bootstrap.py [hunk 2]
```diff
@@ -359,7 +359,7 @@ def validate_localstack_config(name: str):
     docker_env = dict(
         (env.split("=")[0], env.split("=")[1]) for env in ls_service_details.get("environment", {})
     )
-    edge_port = str(docker_env.get("EDGE_PORT") or config.EDGE_PORT)
+    edge_port = str(docker_env.get("EDGE_PORT") or config.GATEWAY_LISTEN[0].port)
```
Comment (joe4dev) [inline localstack/utils/bootstrap.py:362]: Is that `docker_env.get("EDGE_PORT")` a write-usage?

File: bin/docker-entrypoint.sh (hunk 1 above)
Comment (joe4dev) [reply bin/docker-entrypoint.sh:25]: re-added

Comment (simonrw) [reply bin/docker-entrypoint.sh:25]: `LOCALSTACK_HOSTNAME` is not deprecated, it's ignored...

... but the message in `deprecations.py` is wrong. Could you update the messages for the deprecation variables to say "This option has no effect. Please use LOCALSTACK_HOST instead"? After that, then I think we should keep this line so the user gets deprecation warnings, though CLI users will get deprecation warnings twice. Doubly informed I say!

File: localstack/config.py
Comment (simonrw) [inline localstack/config.py:1]: Perhaps we should go the whole hog and remove `EDGE_PORT` completely?

https://github.com/localstack/localstack/pull/9633/files#diff-8ee48b2bcad465473db1a1b202781ca2c7c23db7bd1670e79d45668fb2bd718fR651

**Review by simonrw — APPROVED** [review]
Looks like great changes. I think it's probably a wise choice to remove usages of `EDGE_PORT` completely so we are not tempted by its alluring charms again... :joy:

File: tests/aws/services/apigateway/test_apigateway_basic.py [hunk 3]
```diff
@@ -1217,15 +1218,10 @@ def _get_invoke_endpoint(
...
         if use_hostname:
-            return (
-                f"{proto}://{api_id}.execute-api.{LOCALHOST_HOSTNAME}:{config.EDGE_PORT}/"
-                f"{stage}{path}"
-            )
-        return (
-            f"{proto}://localhost:{config.EDGE_PORT}/restapis/{api_id}/{stage}/_user_request_{path}"
-        )
+            host = f"{api_id}.execute-api.{localstack_host().host}"
+            return f"{config.external_service_url(host=host)}/{stage}{path}"
```
[the remark this answers is not in the document]
Comment (joe4dev) [reply tests/aws/services/apigateway/test_apigateway_basic.py:1222]: agree 💯 

I had exactly the same thought when writing this 👍 
We should add this soon.

File: tests/aws/services/apigateway/test_apigateway_basic.py [hunk 4]
```diff
@@ -1217,15 +1218,10 @@ def _get_invoke_endpoint(
...
+        return f"{config.internal_service_url(host=LOCALHOST)}/restapis/{api_id}/{stage}/_user_request_{path}"
```
[the remark this answers is not in the document]
Comment (joe4dev) [reply tests/aws/services/apigateway/test_apigateway_basic.py:1223]: ups 🤦  Thank you 👍

File: localstack/config.py
Comment (joe4dev) [reply localstack/config.py:1]: that's indeed very tempting 🤩 

I'm not 100% sure whether we have any other (external) dependencies 🤔

Comment (simonrw) [reply localstack/config.py:1]: The same motivation for keeping `get_edge_url` etc. I suppose

Comment (joe4dev) [reply localstack/config.py:1]: I didn't find any usages in the public extensions:
* https://github.com/localstack/localstack-extensions
* Authress Extension: https://github.com/Authress/localstack-extension

Comment (simonrw) [reply localstack/config.py:1]: Scrap it then!

Comment (joe4dev) [reply localstack/config.py:1]: there you go 🧹  https://github.com/localstack/localstack/pull/9636

File: localstack/utils/bootstrap.py (hunk 2 above)
Comment (joe4dev) [reply localstack/utils/bootstrap.py:362]: guess so, removed in the follow-up PR

File: tests/aws/services/apigateway/test_apigateway_basic.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (joe4dev) [reply tests/aws/services/apigateway/test_apigateway_basic.py:1221]: done in https://github.com/localstack/localstack/pull/9646

**Pull request merged**